mypy/newsemanal/semanal.py

```diff
@@ -630,6 +602,9 @@ def analyze_overload_sigs_and_impl(
         types = []
         non_overload_indexes = []
         impl = None  # type: Optional[OverloadPart]
+        if defn.impl:
```

Comment:
Ew the way this works is ugly (the existing logic, not this change in particular). Add TODO item about not modifying the `items` list.